// node-appwrite/dist/services/project.mjs
import { AppwriteException } from '../client.mjs';

// src/services/project.ts
var Project = class {
  constructor(client) {
    this.client = client;
  }
  listKeys(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const apiPath = "/project/keys";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {};
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  createKey(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        keyId: paramsOrFirst,
        name: rest[0],
        scopes: rest[1],
        expire: rest[2]
      };
    }
    const keyId = params.keyId;
    const name = params.name;
    const scopes = params.scopes;
    const expire = params.expire;
    if (typeof keyId === "undefined") {
      throw new AppwriteException('Missing required parameter: "keyId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof scopes === "undefined") {
      throw new AppwriteException('Missing required parameter: "scopes"');
    }
    const apiPath = "/project/keys";
    const payload = {};
    if (typeof keyId !== "undefined") {
      payload["keyId"] = keyId;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof scopes !== "undefined") {
      payload["scopes"] = scopes;
    }
    if (typeof expire !== "undefined") {
      payload["expire"] = expire;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  getKey(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        keyId: paramsOrFirst
      };
    }
    const keyId = params.keyId;
    if (typeof keyId === "undefined") {
      throw new AppwriteException('Missing required parameter: "keyId"');
    }
    const apiPath = "/project/keys/{keyId}".replace("{keyId}", keyId);
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {};
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  updateKey(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        keyId: paramsOrFirst,
        name: rest[0],
        scopes: rest[1],
        expire: rest[2]
      };
    }
    const keyId = params.keyId;
    const name = params.name;
    const scopes = params.scopes;
    const expire = params.expire;
    if (typeof keyId === "undefined") {
      throw new AppwriteException('Missing required parameter: "keyId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof scopes === "undefined") {
      throw new AppwriteException('Missing required parameter: "scopes"');
    }
    const apiPath = "/project/keys/{keyId}".replace("{keyId}", keyId);
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof scopes !== "undefined") {
      payload["scopes"] = scopes;
    }
    if (typeof expire !== "undefined") {
      payload["expire"] = expire;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  deleteKey(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        keyId: paramsOrFirst
      };
    }
    const keyId = params.keyId;
    if (typeof keyId === "undefined") {
      throw new AppwriteException('Missing required parameter: "keyId"');
    }
    const apiPath = "/project/keys/{keyId}".replace("{keyId}", keyId);
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
  updateLabels(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        labels: paramsOrFirst
      };
    }
    const labels = params.labels;
    if (typeof labels === "undefined") {
      throw new AppwriteException('Missing required parameter: "labels"');
    }
    const apiPath = "/project/labels";
    const payload = {};
    if (typeof labels !== "undefined") {
      payload["labels"] = labels;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  listPlatforms(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const apiPath = "/project/platforms";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {};
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  createAndroidPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        applicationId: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const applicationId = params.applicationId;
    if (typeof platformId === "undefined") {
      throw new AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof applicationId === "undefined") {
      throw new AppwriteException('Missing required parameter: "applicationId"');
    }
    const apiPath = "/project/platforms/android";
    const payload = {};
    if (typeof platformId !== "undefined") {
      payload["platformId"] = platformId;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof applicationId !== "undefined") {
      payload["applicationId"] = applicationId;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  updateAndroidPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        applicationId: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const applicationId = params.applicationId;
    if (typeof platformId === "undefined") {
      throw new AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof applicationId === "undefined") {
      throw new AppwriteException('Missing required parameter: "applicationId"');
    }
    const apiPath = "/project/platforms/android/{platformId}".replace("{platformId}", platformId);
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof applicationId !== "undefined") {
      payload["applicationId"] = applicationId;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  createApplePlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        bundleIdentifier: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const bundleIdentifier = params.bundleIdentifier;
    if (typeof platformId === "undefined") {
      throw new AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof bundleIdentifier === "undefined") {
      throw new AppwriteException('Missing required parameter: "bundleIdentifier"');
    }
    const apiPath = "/project/platforms/apple";
    const payload = {};
    if (typeof platformId !== "undefined") {
      payload["platformId"] = platformId;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof bundleIdentifier !== "undefined") {
      payload["bundleIdentifier"] = bundleIdentifier;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  updateApplePlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        bundleIdentifier: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const bundleIdentifier = params.bundleIdentifier;
    if (typeof platformId === "undefined") {
      throw new AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof bundleIdentifier === "undefined") {
      throw new AppwriteException('Missing required parameter: "bundleIdentifier"');
    }
    const apiPath = "/project/platforms/apple/{platformId}".replace("{platformId}", platformId);
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof bundleIdentifier !== "undefined") {
      payload["bundleIdentifier"] = bundleIdentifier;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  createLinuxPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        packageName: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const packageName = params.packageName;
    if (typeof platformId === "undefined") {
      throw new AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof packageName === "undefined") {
      throw new AppwriteException('Missing required parameter: "packageName"');
    }
    const apiPath = "/project/platforms/linux";
    const payload = {};
    if (typeof platformId !== "undefined") {
      payload["platformId"] = platformId;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof packageName !== "undefined") {
      payload["packageName"] = packageName;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  updateLinuxPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        packageName: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const packageName = params.packageName;
    if (typeof platformId === "undefined") {
      throw new AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof packageName === "undefined") {
      throw new AppwriteException('Missing required parameter: "packageName"');
    }
    const apiPath = "/project/platforms/linux/{platformId}".replace("{platformId}", platformId);
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof packageName !== "undefined") {
      payload["packageName"] = packageName;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  createWebPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        hostname: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const hostname = params.hostname;
    if (typeof platformId === "undefined") {
      throw new AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof hostname === "undefined") {
      throw new AppwriteException('Missing required parameter: "hostname"');
    }
    const apiPath = "/project/platforms/web";
    const payload = {};
    if (typeof platformId !== "undefined") {
      payload["platformId"] = platformId;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof hostname !== "undefined") {
      payload["hostname"] = hostname;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  updateWebPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        hostname: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const hostname = params.hostname;
    if (typeof platformId === "undefined") {
      throw new AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof hostname === "undefined") {
      throw new AppwriteException('Missing required parameter: "hostname"');
    }
    const apiPath = "/project/platforms/web/{platformId}".replace("{platformId}", platformId);
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof hostname !== "undefined") {
      payload["hostname"] = hostname;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  createWindowsPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        packageIdentifierName: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const packageIdentifierName = params.packageIdentifierName;
    if (typeof platformId === "undefined") {
      throw new AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof packageIdentifierName === "undefined") {
      throw new AppwriteException('Missing required parameter: "packageIdentifierName"');
    }
    const apiPath = "/project/platforms/windows";
    const payload = {};
    if (typeof platformId !== "undefined") {
      payload["platformId"] = platformId;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof packageIdentifierName !== "undefined") {
      payload["packageIdentifierName"] = packageIdentifierName;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  updateWindowsPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        packageIdentifierName: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const packageIdentifierName = params.packageIdentifierName;
    if (typeof platformId === "undefined") {
      throw new AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof packageIdentifierName === "undefined") {
      throw new AppwriteException('Missing required parameter: "packageIdentifierName"');
    }
    const apiPath = "/project/platforms/windows/{platformId}".replace("{platformId}", platformId);
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof packageIdentifierName !== "undefined") {
      payload["packageIdentifierName"] = packageIdentifierName;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  getPlatform(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst
      };
    }
    const platformId = params.platformId;
    if (typeof platformId === "undefined") {
      throw new AppwriteException('Missing required parameter: "platformId"');
    }
    const apiPath = "/project/platforms/{platformId}".replace("{platformId}", platformId);
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {};
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  deletePlatform(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst
      };
    }
    const platformId = params.platformId;
    if (typeof platformId === "undefined") {
      throw new AppwriteException('Missing required parameter: "platformId"');
    }
    const apiPath = "/project/platforms/{platformId}".replace("{platformId}", platformId);
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
  updateProtocolStatus(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst) && ("protocolId" in paramsOrFirst || "enabled" in paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        protocolId: paramsOrFirst,
        enabled: rest[0]
      };
    }
    const protocolId = params.protocolId;
    const enabled = params.enabled;
    if (typeof protocolId === "undefined") {
      throw new AppwriteException('Missing required parameter: "protocolId"');
    }
    if (typeof enabled === "undefined") {
      throw new AppwriteException('Missing required parameter: "enabled"');
    }
    const apiPath = "/project/protocols/{protocolId}/status".replace("{protocolId}", protocolId);
    const payload = {};
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateServiceStatus(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst) && ("serviceId" in paramsOrFirst || "enabled" in paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        serviceId: paramsOrFirst,
        enabled: rest[0]
      };
    }
    const serviceId = params.serviceId;
    const enabled = params.enabled;
    if (typeof serviceId === "undefined") {
      throw new AppwriteException('Missing required parameter: "serviceId"');
    }
    if (typeof enabled === "undefined") {
      throw new AppwriteException('Missing required parameter: "enabled"');
    }
    const apiPath = "/project/services/{serviceId}/status".replace("{serviceId}", serviceId);
    const payload = {};
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  listVariables(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const apiPath = "/project/variables";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {};
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  createVariable(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        variableId: paramsOrFirst,
        key: rest[0],
        value: rest[1],
        secret: rest[2]
      };
    }
    const variableId = params.variableId;
    const key = params.key;
    const value = params.value;
    const secret = params.secret;
    if (typeof variableId === "undefined") {
      throw new AppwriteException('Missing required parameter: "variableId"');
    }
    if (typeof key === "undefined") {
      throw new AppwriteException('Missing required parameter: "key"');
    }
    if (typeof value === "undefined") {
      throw new AppwriteException('Missing required parameter: "value"');
    }
    const apiPath = "/project/variables";
    const payload = {};
    if (typeof variableId !== "undefined") {
      payload["variableId"] = variableId;
    }
    if (typeof key !== "undefined") {
      payload["key"] = key;
    }
    if (typeof value !== "undefined") {
      payload["value"] = value;
    }
    if (typeof secret !== "undefined") {
      payload["secret"] = secret;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  getVariable(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        variableId: paramsOrFirst
      };
    }
    const variableId = params.variableId;
    if (typeof variableId === "undefined") {
      throw new AppwriteException('Missing required parameter: "variableId"');
    }
    const apiPath = "/project/variables/{variableId}".replace("{variableId}", variableId);
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {};
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  updateVariable(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        variableId: paramsOrFirst,
        key: rest[0],
        value: rest[1],
        secret: rest[2]
      };
    }
    const variableId = params.variableId;
    const key = params.key;
    const value = params.value;
    const secret = params.secret;
    if (typeof variableId === "undefined") {
      throw new AppwriteException('Missing required parameter: "variableId"');
    }
    const apiPath = "/project/variables/{variableId}".replace("{variableId}", variableId);
    const payload = {};
    if (typeof key !== "undefined") {
      payload["key"] = key;
    }
    if (typeof value !== "undefined") {
      payload["value"] = value;
    }
    if (typeof secret !== "undefined") {
      payload["secret"] = secret;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  deleteVariable(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        variableId: paramsOrFirst
      };
    }
    const variableId = params.variableId;
    if (typeof variableId === "undefined") {
      throw new AppwriteException('Missing required parameter: "variableId"');
    }
    const apiPath = "/project/variables/{variableId}".replace("{variableId}", variableId);
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
};

export { Project };
